engines/bops_api/app/jobs/bops_api/planning_application_dependency_job.rb

```diff
@@ -151,7 +153,27 @@ def process_immunity_details(planning_application)
     end
 
     def fetch_constraint_entities(planning_application_constraint, entities)
-      planning_application_constraint.update!(data: fetch_entities(entities))
+      fetched_entities = fetch_entities(entities)
+
+      planning_application_constraint.update!(
+        data: any_error_responses?(fetched_entities) ? [] : fetched_entities,
+        status: any_error_responses?(fetched_entities) ? planning_application_status(fetched_entities) : "success"
+      )
+    end
+
+    def any_error_responses?(fetched_entities)
+      fetched_entities.empty? || fetched_entities.include?("failed") ||
+        fetched_entities.include?("removed") || fetched_entities.include?("not_found")
+    end
+
+    def planning_application_status(fetched_entities)
```

Comment:
I think if this had an `else "success"` at the end, then line 160 could just be `status: planning_application_status(fetched_entities)`.